### PR TITLE
lab 26

### PR DESCRIPTION
**Pull request by loomnugget**

### CODE REVIEW

File: claudia-lab/app/config/router-config.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    },
+  ];
+
+  routes.forEach( route => {
+    $stateProvider.state(route);
+  });
+}
+
+
+// view
```
Comment (bnates) [inline claudia-lab/app/config/router-config.js:39]: @loomnugget this comment block is fine for showing the tree but is best contained on a separate branch, in order to keep `master` production ready

File: claudia-lab/app/config/router-config.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+//   home
+//      home.html
+//      home.js
+//      home.scss
+//   profile
+//      profile.html
+//      profile.scss
+//      profile.js
```
Comment (bnates) [inline claudia-lab/app/config/router-config.js:47]: @loomnugget route config looks good

File: claudia-lab/app/entry.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+// load view controllers in ./view
+context = require.context('./view/', true, /\.js$/);
+context.keys().forEach( key => {
+  let name = pascalcase(path.basename(key, '.js'));
+  console.log('name', name);
+  claudiagram.controller(name, context(key));
+});
+
+// load services
```
Comment (bnates) [inline claudia-lab/app/entry.js:32]: @loomnugget best to pull out the following comment blocks for cleanliness

File: claudia-lab/app/index.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title> My Awesome Startup </title>
+  </head>
+  <body>
+    <h1> My Awesome Startup </h1>
+    <ui-view> </ui-view>
+  </body>
+</html>
```
Comment (bnates) [inline claudia-lab/app/index.html:13]: @loomnugget index looks good

File: claudia-lab/app/scss/lib/_theme.scss [hunk 5]
```diff
@@ -0,0 +1,8 @@
+$brand-black : #000;
+$brand-white : #fff;
+$brand-1st: #7afaf7;
```
Comment (bnates) [inline claudia-lab/app/scss/lib/_theme.scss:3]: @loomnugget using numbers in your `_var` names is fine _but_ my preference is to avoid them - for style points

File: claudia-lab/app/scss/main.scss [hunk 6]
```diff
@@ -0,0 +1,13 @@
+@import "_vendor";
+@import "_theme";
```
Comment (bnates) [inline claudia-lab/app/scss/main.scss:2]: @loomnugget no need to keep the prepended underscore - this gets implied when importing partials:

```
@import "vendor"
@import "theme"
```

File: claudia-lab/app/view/home/home-controller.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+require('./home.scss');
+
+module.exports = ['$log', HomeController];
+
+function HomeController($log){
+  $log.debug('init homeCtrl');
+  this.title = 'Home';
+}
```
Comment (bnates) [inline claudia-lab/app/view/home/home-controller.js:10]: @loomnugget simple but works!

File: claudia-lab/app/view/home/home.scss [hunk 8]
```diff
@@ -0,0 +1,8 @@
+@import "theme";
+
+.home {
+  width: 100%;
+  height: 20%;
+
```
Comment (bnates) [inline claudia-lab/app/view/home/home.scss:6]: @loomnugget no extra line needed here

File: claudia-lab/app/view/profile/profile.scss [hunk 9]
```diff
@@ -0,0 +1,8 @@
+@import "theme";
+
+.profile {
+  width: 100%;
+  height: 20%;
+
+  background-color: $brand-white;
+}
```
Comment (bnates) [inline claudia-lab/app/view/profile/profile.scss:8]: @loomnugget since your profile and home CSS are the same, you could create a partial _or_ a mixin to share styles - for more info on mixins:  https://www.sitepoint.com/sass-basics-the-mixin-directive/

File: claudia-lab/karma.conf.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    reporters: ['progress'],
+    port: 9876,
+    colors: true,
+    logLevel: config.LOG_INFO,
+    autoWatch: true,
+    browsers: ['PhantomJS'],
+    singleRun: false,
+    concurrency: Infinity
+  })
+}
```
Comment (bnates) [inline claudia-lab/karma.conf.js:32]: @loomnugget karma config looks good

File: claudia-lab/test/home-controller-test.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+      this.homeCtrl = new $controller('HomeController');
+    });
+  });
+
+  describe('testing initial properties', () => {
+    it('title should equal "Home"', () => {
+      expect(this.homeCtrl.title).toBe('Home');
+    });
+  });
+});
```
Comment (bnates) [inline claudia-lab/test/home-controller-test.js:16]: @loomnugget tests are minimal but effective - you'll get better, with time, at adding in additional tests for edge cases
